enterprise/tpa_pipeline.py

```diff
@@ -142,10 +142,17 @@ def redirect_to_consent():
     if not enterprise_customer.requests_data_sharing_consent:
         # This enterprise customer attached to this pipeline element does not request data sharing consent;
         # proceed with the creation of a link between the user and the enterprise customer, then exit.
-        EnterpriseCustomerUser.objects.get_or_create(
+        ec_user, _ = EnterpriseCustomerUser.objects.get_or_create(
             enterprise_customer=enterprise_customer,
             user_id=user.id
         )
+
+        if enterprise_customer.enforce_data_sharing_consent == EnterpriseCustomer.EXTERNALLY_MANAGED:
```

Comment:
So if consent is disabled and is not externally managed, do we not create a UserDataSharingConsentAudit record for the EnterpriseCustomerUser? Sorry, that is a lot of NOTs.